sh_scrapy/extension.py

```diff
@@ -79,13 +79,41 @@ def spider_closed(self, spider, reason):
 """
 
 
-class HubstorageMiddleware(object):
+class HubstorageMiddleware:
```

Comment:
@kmike the difference between the two is that the extension writes items, the middleware writes requests

https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/blob/v0.17.4/sh_scrapy/extension.py#L57-L64

Reply:
Ah right, there is extension (HubstorageExtension) and middleware (HubstorageMiddleware) in extension.py module, and it's extension which is enabled, not the middleware! I misread that; there is no issue with writing things twice. 

There is still an issue though, because only HubstorageMiddleware has been updated for centralized fingerprints, not HubstorageDownloaderMiddleware / HubstorageSpiderMiddleware. So, it looks like this PR does nothing to add centralized fingerprints support, as it only modifies deprecated component, which is not enabled by default :(

Reply:
Oh I see what you mean now. I think you're indeed correct, the component that needs to be updated is [sh_scrapy.middlewares.HubstorageDownloaderMiddleware](https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/blob/v0.17.4/sh_scrapy/middlewares.py#L38)

Reply:
Follow up at https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/pull/87